assembl/models/path_utils.py

```diff
@@ -306,17 +321,28 @@ def base_query(labeled=False):
             q = q.alias("relposts")
         return q
 
-    def as_clause(self, db, discussion_id, user_id=None, content=None):
-        subq = self.as_clause_base(db)
+    def as_clause(self, db, discussion_id, user_id=None, content=None,
+                  include_deleted=False):
+        subq = self.as_clause_base(db, include_deleted=include_deleted)
         content = content or with_polymorphic(
             Content, [], Content.__table__,
             aliased=False, flat=True)
 
         q = db.query(content).filter(
                 (content.discussion_id == discussion_id)
                 & (content.hidden == False)
-                & (Content.tombstone_condition(content))
                 ).join(subq, content.id == subq.c.post_id)
+        if include_deleted is not None:
+            if include_deleted:
+                post = with_polymorphic(
+                    Post, [], Post.__table__,
+                    aliased=False, flat=True)
+                q = q.join(
+                    post, (post.id == content.id) &
+                    post.publication_state.in_(deleted_publication_states))
+            else:
+                q = q.filter(content.tombstone_date == None)
+
```

Comment:
@maparent Why is the `with_polymorphic` done on `Post` here rather than `Content`? That is unclear to me.